### DISCUSSION TITLE
Disable the `no-inline-comment` rule

### DISCUSSION THREAD

**Discussion by leroykorterink** [mediamonks/eslint-config#221] [General]
There is a proposal to disable the `no-inline-comments` rule.

> `no-inline-comments:` sometimes inline comments are useful to explain a specific line of code specially when the comment is really short.
> \- @edgardz 

Pleas see the [documentation](https://eslint.org/docs/latest/rules/no-inline-comments)

**Comment by leroykorterink** [comment]
I personally don't use inline comments because I think it's easier to read code when comments are on a new line. Variable names should probably be improved if an inline comment is necessary. I don't really have an opinion about having this rule enabled or not 😅.

**Comment by ThaNarie** [comment]
I've seen this used on two occasions in our current project.

First is when you have short lists or objects, that have short comments after them.
When moving the inline comment above the items, it's makes it way longer, and a bit harder to read/scan.

```ts
export type ApiService =
  | 'bucket' // For static files
  | 'content' // Main content
  | 'identity' // To get app token
  | 'notification' // To listen to SSE events
  | 'search' // ElasticSearch API
  | 'user'; // Technically identity API, but specific for user endpoints that cannot be cached
```

Or when using ternary conditions `?:`, all comments are considered "inline", and the only alternative is to rewrite them as a `let` with an `if/else` assignment.

```
someFunction(
  param1,
  param2,
  isLivePage
    ? // eslint-disable-next-line no-inline-comments
      // Some explanation of why this is used in this case
      isCurrentBlaBlaBla
    : // eslint-disable-next-line no-inline-comments
      // Some other explanation about this specific line
      (): boolean => true
)
```

---

Personally I run into this rule more often than I like, especially in the first case mentioned above (also when defining objects, or passing function params), but mostly when writing some demo code (which admittedly is not in a normal project context).

The problem that might arise when disabling it though, is that people that don't do it "correct" automatically (knowing when better to use or avoid inline comments), will get PR feedback or other people changing their comments later to "improve" it, which is a waste of effort.

So I don't feel strongly about either option, just a slight personal preference for myself.

**Comment by edgardz** [comment]
My personal opinion here as well. The examples above are good.
